addon/initializers/ember-parachute.js

```diff
@@ -11,9 +11,64 @@ const {
   sendEvent
 } = Ember;
 
+const {
+  keys
+} = Object;
+
 export function initialize(/* application */) {
   Ember.Route.reopen({
     /**
+     * Setup the route's `queryParams` map and call the `setup` hook
+     * on the controller.
+     *
+     * @method setupController
+     * @public
+     * @param {Ember.Controller} controller
+     * @returns {void}
+     */
+    setupController(controller) {
+      this._super(...arguments);
+
+      if (QueryParams.hasParachute(controller)) {
+        this._setupParachuteQueryParamsMap(controller);
+
+        let { routeName } = this;
+        let event = new ParachuteEvent(routeName, controller, {});
+
+        // Overrides
+        event.changed = event.changes;
+        event.shouldRefresh = true;
+
+        tryInvoke(controller, 'setup', [event]);
+        sendEvent(controller, 'setup', [event]);
+      }
+    },
+
+    /**
+     * Call the `reset` hook on the controller.
+     *
+     * @method resetController
+     * @public
+     * @param {Ember.Controller} controller
+     * @param  {Boolean} isExiting
+     * @returns {void}
+     */
+    resetController(controller, isExiting) {
+      this._super(...arguments);
+
+      if (QueryParams.hasParachute(controller)) {
+        let { routeName } = this;
+        let event = new ParachuteEvent(routeName, controller, {});
+
+        // Overrides
+        event.shouldRefresh = false;
+
+        tryInvoke(controller, 'reset', [event, isExiting]);
```

Comment:
I added it for the sake of consistency. That way you have access to all the same data in `setup`, `reset`, and `queryParamsDidChange`.